www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Nginx.xml

```diff
@@ -662,14 +662,24 @@
         <Required>N</Required>
         <multiple>Y</multiple>
       </syslog_targets>
-      <listen_http_port type="PortField">
+      <listen_http_address type="CSVListField">
         <Required>N</Required>
-        <default>80</default>
-      </listen_http_port>
-      <listen_https_port type="PortField">
+        <multiple>Y</multiple>
+        <default>80,[::]:80</default>
+        <mask>/^(?:(?:(?:\d{1,3}(?:\.\d{1,3}){3})|(?:\[[a-f0-9:]{1,4}(?::[a-f0-9:]{0,4}){1,7}\])):\d+|:?\d+)(?:\s*,\s*(?:(?:(?:\d{1,3}(?:\.\d{1,3}){3})|(?:\[[a-f0-9:]{1,4}(?::[a-f0-9:]{0,4}){1,7}\])):\d+|:?\d+))*$/i</mask>
+        <ValidationMessage>Please provide a valid listen address or port, i.e. 127.0.0.1:8080, [::1]:8080, 8080.</ValidationMessage>
+      </listen_http_address>
+      <listen_https_address type="CSVListField">
         <Required>N</Required>
-        <default>443</default>
-      </listen_https_port>
+        <multiple>Y</multiple>
+        <default>443,[::]:443</default>
+        <mask>/^(?:(?:(?:\d{1,3}(?:\.\d{1,3}){3})|(?:\[[a-f0-9:]{1,4}(?::[a-f0-9:]{0,4}){1,7}\])):\d+|:?\d+)(?:\s*,\s*(?:(?:(?:\d{1,3}(?:\.\d{1,3}){3})|(?:\[[a-f0-9:]{1,4}(?::[a-f0-9:]{0,4}){1,7}\])):\d+|:?\d+))*$/i</mask>
+        <ValidationMessage>Please provide a valid listen address or port, i.e. 127.0.0.1:8080, [::1]:8080, 8080.</ValidationMessage>
+      </listen_https_address>
+      <default_server type="BooleanField">
+        <Required>Y</Required>
+        <default>0</default>
+      </default_server>
```

Comment:
this needs some kind of a unique constraint. There are likely already some examples how to do custom validations in this plugin.

Completely custom validation plugin:
https://github.com/opnsense/plugins/blob/master/www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php

Here are some core contraints, that are available as well:
https://github.com/opnsense/core/tree/master/src/opnsense/mvc/app/models/OPNsense/Base/Constraints